pkg/kubelet/server/server_test.go

```diff
@@ -1112,9 +1107,6 @@ func testExecAttach(t *testing.T, verb string) {
 			require.NoError(t, err, "POSTing")
 			defer resp.Body.Close()
 
-			_, err = io.ReadAll(resp.Body)
-			assert.NoError(t, err, "reading response body")
-
```

Comment:
Why was this removed?

Reply:
Today, we create this response "manually" with http.ReadResponse here: https://github.com/kubernetes/kubernetes/blob/f87231003aeb43bf884cf9dc10b1247b8ae5cbb8/staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper.go#L317

The body is http.NoBody because the status is 101 switching protocols: https://cs.opensource.google/go/go/+/master:src/net/http/transfer.go;l=561;drc=3a41094107237535673bdbbdaabc10e54cb2b67c

In this PR, the response is now created by http.Transport. For 101 switching protocols, the transport sets up the body after calling http.ReadResponse: https://cs.opensource.google/go/go/+/master:src/net/http/transport.go;l=2319-2321;drc=3a41094107237535673bdbbdaabc10e54cb2b67c

So now, resp.Body is something that we want to leave open while we're streaming.